pages/[envId]/[slug].tsx

```diff
@@ -48,41 +49,48 @@ export const getStaticProps: GetStaticProps<Props, IParams> = async (context) =>
 
   const previewApiKey = getPreviewApiKeyFromPreviewData(context.previewData);
 
-  const siteMenu = await getSiteMenu({ envId, previewApiKey }, !!context.preview);
+  const siteMenuData = await getSiteMenu({ envId, previewApiKey }, !!context.preview);
   const defaultMetadata = await getDefaultMetadata({ envId, previewApiKey }, !!context.preview);
 
-  const page = await getItemBySlug<WSL_Page>({ envId, previewApiKey }, slug, contentTypes.page.codename, !!context.preview);
+  const pageData = await getItemBySlug<WSL_Page>({ envId, previewApiKey }, slug, contentTypes.page.codename, !!context.preview);
 
-  if (page === null) {
+  if (pageData === null) {
     return {
       notFound: true
     };
   }
 
+  const page = stringifyAsType(pageData);

```

Comment:
I don't understand this. Why do I only send a string with the type definition to the client, and then let him parse data back to the object? isn't it easier to do the stringify and parsing here at the backend and send the client just what it needs? Or am I missing something?

Reply:
getStaticProps serializes the values you return in it. if they have circular references, the app will crash. instead, we first flatten the object into a string format, then restore its data client-side, thus avoiding any circular issues in getStaticProps without losing any information from the object.